fs/operations/operations.go

```diff
@@ -522,6 +512,21 @@ func Copy(ctx context.Context, f fs.Fs, dst fs.Object, remote string, src fs.Obj
 			return newDst, err
 		}
 	}
+
+	// Move the copied file to its real destination.
+	if remotePartial != remote && err == nil {
+		dst, err = f.Features().Move(ctx, newDst, remote)
+		if err == nil {
+			fs.Debugf(newDst, "renamed to: %s", remote)
+			newDst = dst
+		} else {
+			fs.Errorf(newDst, "partial file rename failed: %v", err)
+			err = fs.CountError(err)
+			removeFailedCopy(ctx, newDst)
+			return newDst, err
```

Comment:
I'm a little confused why there's `dst` and `newDst` in this function. It might be because there are a lot of assignments to `dst` and `newDst` is updated to "carry the newly created file handle forward" when there were no errors.  

I think the error handling here is correct and the debug prints print sensible output too.  I'm just not 100% sure what should be returned from this path and how the `newDst` is used by the caller.

Reply:
I think this looks correct.

The return from this function should be the final object in its final resting place. We try not to update `newDst` unless the function returned `err = nil` which is why I think there is `dst` and `newDst`.

I think the whole `dst` and `newDst` needs a careful review, but let's not do it in this PR! I'll do it after we merge it.